drivers/pinctrl/pinctrl_ite_it8xxx2.c

```diff
@@ -18,8 +18,13 @@ LOG_MODULE_REGISTER(pinctrl_ite_it8xxx2, LOG_LEVEL_ERR);
 #define GPIO_GROUP_MEMBERS  8
 
 struct pinctrl_it8xxx2_config {
-	/* gpio port control register (byte mapping to pin) */
+	/*
+	 * gpio port control register (byte mapping to pin), or
+	 * KSI[7:0]/KSO[15:8]/KSO[7:0] port gpio control register (bit mapping to pin)
+	 */
```

Comment:
Can this structure get organized as a union to save space?

```
struct it8xxx2_gpio_pinctrl {
    uint8_t *reg_gpcr;
    uintptr_t func3_gcr[GPIO_GROUP_MEMBERS];
    /* other regs only used by GPIO pinctrl */
};
struct it8xxx_ksi_kso_pinctrl {
    uint8_t *reg_gctrl;
    uint8_t *reg_ctrl;
};
struct pinctrl_it8xxx2_config {
    enum it8xxx_pinctrl_type type;
    union {
        struct it8xxx2_gpio_pinctrl gpio;
        struct it8xxx2_ksi_kso_pinctrl ksi_kso;
    };
```
You would use the `gpio-group` property to determine whether to initialize `pinctrl_it8xxx2_config.gpio` or `pinctrl_it8xxx2_config.ksi_kso`.

If there are any properties required by both GPIO and KSI/KSO, then add those as common fields outside the union.

Reply:
Done